libs/barista-components/sunburst/src/sunburst.ts

```diff
@@ -0,0 +1,261 @@
+} from '@angular/core';
+import {
+  dtSetUiTestAttribute,
+  DtUiTestConfiguration,
+  DT_UI_TEST_CONFIG,
+} from '@dynatrace/barista-components/core';
+import { DtSunburstOverlay } from './sunburst.directive';
+import {
+  DtSunburstNode,
+  DtSunburstNodeInternal,
+  DtSunburstSlice,
+  DtSunburstValueMode,
+  fillNodes,
+  getNodesWithState,
+  getSelectedId,
+  getSelectedNodes,
+  getSelectedNodesFromOutside,
+  getSlices,
+  getValue,
+} from './sunburst.util';
+
+const OVERLAY_PANEL_CLASS = 'dt-sunburst-overlay-panel';
+
+@Component({
+  selector: 'dt-sunburst',
+  templateUrl: 'sunburst.html',
+  styleUrls: ['sunburst.scss'],
+  preserveWhitespaces: false,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.Emulated,
+})
+export class DtSunburst implements OnChanges {
+  readonly width = 480;
+  readonly viewBox = '-240 -176 480 352';
+
+  @ViewChild('svg') svgEl;
+
+  /** @internal Template reference for the DtSunburst overlay. */
+  @ContentChild(DtSunburstOverlay, { read: TemplateRef })
+  // tslint:disable-next-line: no-any
+  private _overlay: TemplateRef<any>;
+
+  filledSeries: DtSunburstNodeInternal[];
+  slices: DtSunburstSlice[];
+
+  _selected: DtSunburstNodeInternal[];
+  selectedLabel: string;
+  selectedValue: number;
+  selectedRelativeValue: number;
+  labelAsAbsolute: boolean = true;
+
+  @HostListener('click', ['$event']) onClick(ev?: MouseEvent): void {
+    this.select(ev);
+  }
+
+  @Input() series: DtSunburstNode[];
+  @Input() selected: DtSunburstNode[];
+  @Input() noSelectionLabel: string = 'All';
+  @Input() valueDisplayMode: DtSunburstValueMode = DtSunburstValueMode.ABSOLUTE;
+
+  @Output() selectedChange: EventEmitter<DtSunburstNode[]> = new EventEmitter();
+
+  /** Template portal for the default overlay */
+  // tslint:disable-next-line: use-default-type-parameter no-any
+  private _portal: TemplatePortal<any> | null;
+
+  /** Reference to the open overlay. */
+  private _overlayRef: OverlayRef;
+
+  constructor(
+    private _overlayService: Overlay,
+    private _viewContainerRef: ViewContainerRef,
+    private _elementRef?: ElementRef<HTMLElement>,
+    @Optional()
+    @Inject(DT_UI_TEST_CONFIG)
+    private _config?: DtUiTestConfiguration,
+  ) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.series?.currentValue !== changes.series?.previousValue) {
+      this.filledSeries = fillNodes(this.series);
+      this.render();
+    }
+
+    if (changes.selected?.currentValue !== changes.selected?.previousValue) {
+      this._selected = getSelectedNodesFromOutside(
+        this.filledSeries,
+        this.selected,
+      );
+      this.render();
+    }
+
+    if (
+      changes.valueDisplayMode?.currentValue !==
+      changes.valueDisplayMode?.previousValue
+    ) {
+      this.labelAsAbsolute =
+        this.valueDisplayMode === DtSunburstValueMode.ABSOLUTE;
+    }
+  }
+
+  select(event?: MouseEvent, slice?: DtSunburstSlice): void {
```

Comment:
Can be marked as an internal. I don't think that the way you use select in the template that event is optional. 
```suggestion
  select(event: MouseEvent, slice?: DtSunburstSlice): void {
```